### PR TITLE
feat(multitenant)! Account prefixed API topics and manifest storage

### PR DESCRIPTION
**Pull request by brooksmtownsend**
## Feature or Problem
This PR ensures we are storing and querying manifests based on an account prefix if running in multitenant mode. This ensures that accounts can only access and modify manifests that are specifically bound to that account, rather than simply using a global lattice ID.

Due to possible edge cases, it's still recommended to keep lattice prefixes globally unique between accounts, as there are not e2e tests at this time to verify that functionality works without issues. This PR at least implements the portion of that that constrains manifests from conflicting between accounts.

## Related Issues
#132

## Release Information
v0.5.0

## Consumer Impact
Multitenancy hasn't released yet so this shouldn't affect consumers, however they will need to examine the account prefixed topic if manually looking up manifests in NATS.

## Testing

Built on platform(s)
- [ ] x86_64-linux
- [ ] aarch64-linux
- [ ] x86_64-darwin
- [x] aarch64-darwin
- [ ] x86_64-windows

Tested on platform(s)
- [ ] x86_64-linux
- [ ] aarch64-linux
- [ ] x86_64-darwin
- [x] aarch64-darwin
- [ ] x86_64-windows

### Unit Test(s)
Modified a few unit tests that were hitting deprecation warnings

### Acceptance or Integration
Will have to modify the e2e tests to properly import/export account prefixes

### Manual Verification
Using the multitenant docker compose, I was able to verify that if I put a manifest using one account, I cannot query that manifest in another account even if I use the correct lattice prefix 😄

### CODE REVIEW

File: src/server/mod.rs [hunk 1]
```diff
@@ -51,7 +53,15 @@ impl<P: Publisher> Server<P> {
...
 
-        let topic = format!("{prefix}.>");
+        let topic_prefix = if multitenant {
+            format!("*.{prefix}")
+        } else {
+            prefix.clone()
+        };
+
+        let topic = format!("{topic_prefix}.>");
+        eprintln!("GONNA LISTEN ON: {topic}");
```
Comment (connorsmith256) [inline src/server/mod.rs:63]: Oopsie whoopsie

File: src/server/mod.rs [hunk 2]
```diff
@@ -81,7 +92,7 @@ impl<P: Publisher> Server<P> {
     #[instrument(level = "info", skip_all)]
     pub async fn serve(mut self) -> anyhow::Result<()> {
         while let Some(msg) = self.subscriber.next().await {
-            if !msg.subject.starts_with(&self.prefix) {
+            if !msg.subject.starts_with(&self.prefix) && !self.multitenant {
                 warn!(subject = %msg.subject, "Received message on an invalid subject");
```
Comment (connorsmith256) [inline src/server/mod.rs:96]: When would this happen anyway? I assume the subscription was always on `{prefix}.>` before, and now can also be `*.{prefix}.>`

File: src/server/mod.rs [hunk 3]
```diff
@@ -196,6 +252,7 @@ impl<P: Publisher> Server<P> {
             anyhow::bail!("Found extra components of subject")
         }
         Ok(ParsedSubject {
+            account_id: account_id,
```
Comment (connorsmith256) [inline src/server/mod.rs:255]: Nit: just `account_id`?

**Review by connorsmith256 — APPROVED** [review]

File: src/server/mod.rs (hunk 1 above)
Comment (brooksmtownsend) [reply src/server/mod.rs]: I had full confidence and definitely never needed to verify this

**Review by connorsmith256 — APPROVED** [review]
